src/Driver/PDO/Statement.php

```diff
@@ -117,12 +133,109 @@ public function execute($params = null): ResultInterface
             );
         }
 
+        $resourceOffsets = $this->getResourceOffsets();
         try {
             $this->stmt->execute($params);
         } catch (PDOException $exception) {
             throw Exception::new($exception);
+        } finally {
+            if ($resourceOffsets !== null) {
+                $this->restoreResourceOffsets($resourceOffsets);
+            }
         }
 
         return new Result($this->stmt);
     }
+
+    /**
+     * Track a binary parameter reference at binding time. These
+     * are cached for later analysis by the getResourceOffsets.
+     *
+     * @param mixed $resource
+     */
+    private function trackParamResource($resource): void
+    {
+        if (! is_resource($resource)) {
+            return;
+        }
+
+        $resources   = &$this->paramResources;
+        $resources ??= [];
+        $resources[] = $resource;
+    }
+
+    /**
+     * Determine the offset that any resource parameters needs to be
+     * restored to after the statement is executed. Call immediately
+     * before execute (not during bindValue) to get the most accurate offset.
+     *
+     * @return int[]|null Return offsets to restore if needed.
+     */
+    private function getResourceOffsets()
+    {
+        $resources = &$this->paramResources;
+        if ($resources === null) {
+            return null;
+        }
+
+        $count           = count($resources);
+        $resourceOffsets = null;
+        $deref           = [];
+        for ($i = 0; $i < $count; ++$i) {
+            $resource = $resources[$i];
+            $position = false;
+            try {
+                if (stream_get_meta_data($resource)['seekable']) {
+                    $position = ftell($resource);
+                }
+            } catch (Throwable $e) {
+                // Swallow
```

Comment:
Again: Unless we have a _very_ good reason to ignore all `Throwable`s, we should not do it.

Reply:
Same comment as with IBMDB2/Statement.php earlier